src/ert/run_models/everest_run_model.py

```diff
@@ -703,3 +716,42 @@ def _handle_errors(
             self._fm_errors[error_hash]["ids"].append(fm_id)
             error_id = self._fm_errors[error_hash]["error_id"]
             fm_logger.error(err_msg.format(error_id, ""))
+
+
+class SimulatorCache:
```

Comment:
Just a general question maybe not related to this PR: Do we still need a simulator cache? It stores some controls and resulting constraints/objectives. I guess also when we have everest storage we could get it from there instead?

Reply:
Yes, we do need it, some optimizers may repeatedly ask for the same value. It would be a good idea to get is somehow from storage. Let's look into that when the storage has stabilized.

Reply:
The controls are vectors of float values, so not hashable, hence `lru_cache `is not straigtforward to use. They do not have some kind of hashable  ID, I think numpy arrays don't have that.

For each control vector we will have multiple evaluations, i.e. the different realizations, hence the additional use of the realization names in the cache.